tests/test_modeling_tf_wav2vec2.py

```diff
@@ -0,0 +1,349 @@
+
+        test = np.random.rand(1, 16000).astype(np.float32)
+
+        tf_input = test
+        pt_input = torch.from_numpy(test)
+
+        tf_output = tf_layer(tf_input).last_hidden_state.numpy()
+        pt_output = pt_layer(pt_input).last_hidden_state.detach().numpy()
+
+        tf.testing.assert_near(tf_output, pt_output, atol=1e-3)
+
+    def check_feed_forward(self):
+        config = Wav2Vec2Config()
+        for i in range(config.num_hidden_layers):
+            tf_layer = self.tf_model.wav2vec2.encoder.layer[i].feed_forward
+            pt_layer = self.pt_model.wav2vec2.encoder.layers[i].feed_forward
+
+            test = np.random.rand(1, 292, 768).astype(np.float32)
+
+            tf_input = test
+            pt_input = torch.from_numpy(test)
+
+            tf_output = tf_layer(tf_input)[0].numpy()
+            pt_output = pt_layer(pt_input)[0].detach().numpy()
+
+            tf.testing.assert_near(tf_output, pt_output, atol=1e-3)
+
+    def check_ctc_head(self):
+        tf_layer = self.tf_model.lm_head
+        pt_layer = self.pt_model.lm_head
+
+        test = np.random.rand(1, 292, 768).astype(np.float32)
+
+        tf_input = test
+        pt_input = torch.from_numpy(test)
+
+        tf_output = tf_layer(tf_input).numpy()
+        pt_output = pt_layer(pt_input).detach().numpy()
+
+        tf.testing.assert_near(tf_output, pt_output, atol=1e-3)
+
+    def check_encoder(self):
+        tf_layer = self.tf_model.wav2vec2.encoder
+        pt_layer = self.pt_model.wav2vec2.encoder
+
+        test = np.random.rand(1, 292, 768).astype(np.float32)
+
+        tf_input = test
+        pt_input = torch.from_numpy(test)
+
+        tf_output = tf_layer(tf_input).last_hidden_state.numpy()
+        pt_output = pt_layer(pt_input).last_hidden_state.detach().numpy()
+
+        tf.testing.assert_near(tf_output, pt_output, atol=1e-3)
+
+    def create_and_check_batch_inference(self, config, input_values, *args):
+        # test does not pass for models making use of `group_norm`
+        # check: https://github.com/pytorch/fairseq/issues/3227
+        pass
+
+    def prepare_config_and_inputs_for_common(self):
+        config, inputs_dict = self.prepare_config_and_inputs()
+        return config, inputs_dict
+
+
+@require_torch
+@require_tf
+class TFWav2Vec2ModelTest(TFModelTesterMixin, unittest.TestCase):
+
+    all_model_classes = (TFWav2Vec2Model, TFWav2Vec2ForCTC) if is_tf_available() else ()
+    test_resize_embeddings = False
+    test_head_masking = False
+    test_onnx = False
+
+    def setUp(self):
+        self.model_tester = TFWav2Vec2ModelTester(self)
+        self.config_tester = ConfigTester(self, config_class=Wav2Vec2Config, hidden_size=37)
+
+    def test_config(self):
+        self.config_tester.run_common_tests()
+
+    def test_model(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_model(*config_and_inputs)
+
+    def test_hidden_states_output(self):
```

Comment:
Think it should be relatively easy to make this work no? Would be great to add the functionality here :-)

Reply:
I ported this from the common tf version, but from what I can tell the pytorch tests works because of [this](https://github.com/huggingface/transformers/blob/7630c11f326025ef67a99db913de9fafcfc0704d/tests/test_modeling_common.py#L939) part.